bin/node/node/src/service.rs

```diff
@@ -36,137 +37,175 @@ native_executor_instance!(
-			Ok(sc_transaction_pool::BasicPool::new(
-				builder.config().transaction_pool.clone(),
-				std::sync::Arc::new(pool_api),
-				builder.prometheus_registry(),
-			))
-		})?
-		.with_import_queue(
-			|_config, client, mut select_chain, _transaction_pool, spawn_task_handle, registry| {
-				let select_chain = select_chain
-					.take()
-					.ok_or_else(|| sc_service::Error::SelectChainRequired)?;
-
-				let (grandpa_block_import, grandpa_link) =
-					grandpa::block_import(client.clone(), &(client.clone() as Arc<_>), select_chain)?;
-
-				let aura_block_import = sc_consensus_aura::AuraBlockImport::<_, _, _, AuraPair>::new(
-					grandpa_block_import.clone(),
-					client.clone(),
-				);
-
-				let import_queue = sc_consensus_aura::import_queue::<_, _, _, AuraPair, _>(
-					sc_consensus_aura::slot_duration(&*client)?,
-					aura_block_import,
-					Some(Box::new(grandpa_block_import.clone())),
-					None,
-					client,
-					inherent_data_providers.clone(),
-					spawn_task_handle,
-					registry,
-				)?;
-
-				import_setup = Some((grandpa_block_import, grandpa_link));
-
-				Ok(import_queue)
-			},
-			)?
-		.with_rpc_extensions(|builder| -> Result<jsonrpc_core::IoHandler<sc_rpc::Metadata>, _> {
-			use sc_rpc::DenyUnsafe;
-			use substrate_frame_rpc_system::{FullSystem, SystemApi};
-
+type FullClient = sc_service::TFullClient<Block, RuntimeApi, Executor>;
+type FullBackend = sc_service::TFullBackend<Block>;
+type FullSelectChain = sc_consensus::LongestChain<FullBackend, Block>;
+
+#[allow(clippy::type_complexity)]
+pub fn new_full_params(
+	config: Configuration,
+) -> Result<
+	(
+		sc_service::ServiceParams<
+			Block,
+			FullClient,
+			sc_consensus_aura::AuraImportQueue<Block, FullClient>,
+			sc_transaction_pool::FullPool<Block, FullClient>,
+			jsonrpc_core::IoHandler<sc_rpc::Metadata>,
+			FullBackend,
+		>,
+		FullSelectChain,
+		sp_inherents::InherentDataProviders,
+		sc_finality_grandpa::GrandpaBlockImport<FullBackend, Block, FullClient, FullSelectChain>,
+		sc_finality_grandpa::LinkHalf<Block, FullClient, FullSelectChain>,
+	),
+	ServiceError,
+> {
+	let inherent_data_providers = sp_inherents::InherentDataProviders::new();
+
+	let (client, backend, keystore, task_manager) = sc_service::new_full_parts::<Block, RuntimeApi, Executor>(&config)?;
+	let client = Arc::new(client);
+
+	let select_chain = sc_consensus::LongestChain::new(backend.clone());
+
+	let pool_api = sc_transaction_pool::FullChainApi::new(client.clone(), config.prometheus_registry());
+	let transaction_pool = sc_transaction_pool::BasicPool::new_full(
+		config.transaction_pool.clone(),
+		std::sync::Arc::new(pool_api),
+		config.prometheus_registry(),
+		task_manager.spawn_handle(),
+		client.clone(),
+	);
+
+	let (grandpa_block_import, grandpa_link) =
+		sc_finality_grandpa::block_import(client.clone(), &(client.clone() as Arc<_>), select_chain.clone())?;
+
+	let aura_block_import =
```

Comment:
I didn't notice that earlier, but we run with `AuRa + Grandpa`? That's pretty cool and hopefully works :)